modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/java/com/liferay/dynamic/data/mapping/upgrade/v3_7_3/test/UpgradeDDMFormInstanceReportTest.java

```diff
@@ -0,0 +1,547 @@
+
+		checkboxMultipleDDMFormFieldOptions.addOptionLabel(
+			"option0", LocaleUtil.US, "option0");
+		checkboxMultipleDDMFormFieldOptions.addOptionLabel(
+			"option1", LocaleUtil.US, "option1");
+		checkboxMultipleDDMFormFieldOptions.addOptionLabel(
+			"option2", LocaleUtil.US, "option2");
+		checkboxMultipleDDMFormFieldOptions.addOptionLabel(
+			"option3", LocaleUtil.US, "option3");
+		checkboxMultipleDDMFormFieldOptions.addOptionLabel(
+			"option4", LocaleUtil.US, "option4");
+
+		ddmForm.addDDMFormField(checkboxMultipleDDMFormField);
+
+		DDMFormField gridFormField = DDMFormTestUtil.createDDMFormField(
+			"grid", "grid", "grid", "string", false, false, false);
+
+		DDMFormFieldOptions gridRowDDMFormFieldOptions =
+			new DDMFormFieldOptions();
+
+		gridRowDDMFormFieldOptions.addOptionLabel(
+			"row0", LocaleUtil.US, "row0");
+		gridRowDDMFormFieldOptions.addOptionLabel(
+			"row1", LocaleUtil.US, "row1");
+		gridRowDDMFormFieldOptions.addOptionLabel(
+			"row2", LocaleUtil.US, "row2");
+		gridRowDDMFormFieldOptions.addOptionLabel(
+			"row3", LocaleUtil.US, "row3");
+		gridRowDDMFormFieldOptions.addOptionLabel(
+			"row4", LocaleUtil.US, "row4");
+
+		gridFormField.setProperty("rows", gridRowDDMFormFieldOptions);
+
+		DDMFormFieldOptions gridColumnDDMFormFieldOptions =
+			new DDMFormFieldOptions();
+
+		gridColumnDDMFormFieldOptions.addOptionLabel(
+			"column0", LocaleUtil.US, "column0");
+		gridColumnDDMFormFieldOptions.addOptionLabel(
+			"column1", LocaleUtil.US, "column1");
+		gridColumnDDMFormFieldOptions.addOptionLabel(
+			"column2", LocaleUtil.US, "column2");
+		gridColumnDDMFormFieldOptions.addOptionLabel(
+			"column3", LocaleUtil.US, "column3");
+		gridColumnDDMFormFieldOptions.addOptionLabel(
+			"column4", LocaleUtil.US, "column4");
+
+		gridFormField.setProperty("columns", gridColumnDDMFormFieldOptions);
+
+		ddmForm.addDDMFormField(gridFormField);
+
+		ddmForm.addDDMFormField(
+			DDMFormTestUtil.createDDMFormField(
+				"numeric", "numeric", "numeric", FieldConstants.DOUBLE, false,
+				false, false));
+
+		DDMFormField radioDDMFormField = DDMFormTestUtil.createDDMFormField(
+			"radio", "radio", "radio", "string", false, false, false);
+
+		DDMFormFieldOptions radioDDMFormFieldOptions =
+			radioDDMFormField.getDDMFormFieldOptions();
+
+		radioDDMFormFieldOptions.addOptionLabel(
+			"option0", LocaleUtil.US, "option0");
+		radioDDMFormFieldOptions.addOptionLabel(
+			"option1", LocaleUtil.US, "option1");
+		radioDDMFormFieldOptions.addOptionLabel(
+			"option2", LocaleUtil.US, "option2");
+		radioDDMFormFieldOptions.addOptionLabel(
+			"option3", LocaleUtil.US, "option3");
+		radioDDMFormFieldOptions.addOptionLabel(
+			"option4", LocaleUtil.US, "option4");
+
+		ddmForm.addDDMFormField(radioDDMFormField);
+
+		return ddmForm;
+	}
+
+	private void _createDDMFormInstanceRecords(DDMFormInstance ddmFormInstance)
+		throws Exception {
+
+		DDMForm ddmForm = _createDDMForm();
+
+		int qtd = 0;
```

Comment:
@marcosapmf, is `qtd` an acronym for `quantity`? Would be better we use something like `size` ou `length`. 😉